romancal/skymatch/skymatch_step.py

```diff
@@ -0,0 +1,218 @@
+"""
+Roman step for sky matching.
+"""
+
+import logging
+from copy import deepcopy
+from itertools import chain
+
+import numpy as np
+from astropy.nddata.bitmask import bitfield_to_boolean_mask, interpret_bit_flags
+from roman_datamodels import datamodels as rdd
+
+from romancal.lib.dqflags import pixel
+from romancal.stpipe import RomanStep
+
+from .skyimage import SkyImage
+from .skymatch import match
+from .skystatistics import SkyStats
+
+__all__ = ["SkyMatchStep"]
+
+
+class SkyMatchStep(RomanStep):
+    """
+    SkyMatchStep: Subtraction or equalization of sky background in science images.
+    """
+
+    class_alias = "skymatch"
+
+    spec = """
+        # General sky matching parameters:
+        skymethod = option('local', 'global', 'match', 'global+match', default='match') # sky computation method
+        match_down = boolean(default=True) # adjust sky to lowest measured value?
+        subtract = boolean(default=False) # subtract computed sky from image data?
+
+        # Image's bounding polygon parameters:
+        stepsize = integer(default=None) # Max vertex separation
+
+        # Sky statistics parameters:
+        skystat = option('median', 'midpt', 'mean', 'mode', default='mode') # sky statistics
+        dqbits = string(default='~DO_NOT_USE+NON_SCIENCE') # "good" DQ bits
+        lower = float(default=None) # Lower limit of "good" pixel values
+        upper = float(default=None) # Upper limit of "good" pixel values
+        nclip = integer(min=0, default=5) # number of sky clipping iterations
+        lsigma = float(min=0.0, default=4.0) # Lower clipping limit, in sigma
+        usigma = float(min=0.0, default=4.0) # Upper clipping limit, in sigma
+        binwidth = float(min=0.0, default=0.1) # Bin width for 'mode' and 'midpt' `skystat`, in sigma
+    """  # noqa: E501
+
+    reference_file_types = []
+
+    def process(self, input):
+        self.log.setLevel(logging.DEBUG)
+        # for now turn off memory optimization until we have better machinery
+        # to handle outputs in a consistent way.
+        self._is_asn = False
```

Comment:
Where is the memory optimization?

Reply:
This was recently implemented in `jwst` by @mcara, but not implemented here. I'll remove this comment.